src/distribution/pareto.rs

```diff
@@ -0,0 +1,539 @@
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// INF
+    /// ```
+    fn max(&self) -> f64 {
+        f64::INFINITY
+    }
+}
+
+impl Mean<f64> for Pareto {
+    /// Returns the mean of the Pareto distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// if α <= 1 {
+    ///     INF
+    /// } else {
+    ///     (α * x_m)/(α - 1)
+    /// }
+    /// ```
+    ///
+    /// where `x_m` is the scale and `α` is the shape
+    fn mean(&self) -> f64 {
+        if self.shape <= 1.0 {
+            f64::INFINITY
+        } else {
+            (self.shape * self.scale) / (self.shape - 1.0)
+        }
+    }
+}
+
+impl Variance<f64> for Pareto {
+    /// Returns the variance of the Pareto distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// if α <= 2 {
+    ///     INF
+    /// } else {
+    ///     (x_m/(α - 1))^2 * (α/(α - 2))
+    /// }
+    /// ```
+    ///
+    /// where `x_m` is the scale and `α` is the shape
+    fn variance(&self) -> f64 {
+        if self.shape <= 2.0 {
+            f64::INFINITY
+        } else {
+            let a = self.scale / (self.shape - 1.0); // just a temporary variable
+            a * a * self.shape / (self.shape - 2.0)
+        }
+    }
+
+    /// Returns the standard deviation of the Pareto distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// let variance = if α <= 2 {
+    ///     INF
+    /// } else {
+    ///     (x_m/(α - 1))^2 * (α/(α - 2))
+    /// };
+    /// sqrt(variance)
+    /// ```
+    ///
+    /// where `x_m` is the scale and `α` is the shape
+    fn std_dev(&self) -> f64 {
+        self.variance().sqrt()
+    }
+}
+
+impl Entropy<f64> for Pareto {
+    /// Returns the entropy for the Pareto distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// ln(α/x_m) - 1/α - 1
+    /// ```
+    ///
+    /// where `x_m` is the scale and `α` is the shape
+    fn entropy(&self) -> f64 {
+        self.shape.ln() - self.scale.ln() - (1.0 / self.shape) - 1.0
+    }
+}
+
+impl Skewness<f64> for Pareto {
+    /// Returns the skewness of the Pareto distribution
+    ///
+    /// # Panics
+    ///
+    /// If `α <= 3.0`
+    ///
+    /// where `α` is the shape
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    ///     (2*(α + 1)/(α - 3))*sqrt((α - 2)/α)
+    /// ```
+    ///
+    /// where `α` is the shape
+    fn skewness(&self) -> f64 {
+        assert!(
+            self.shape > 3.0,
+            format!("{}", StatsError::ArgGt("shape", 3.0))
+        );
+        (2.0 * (self.shape + 1.0) / (self.shape - 3.0)) * ((self.shape - 2.0) / self.shape).sqrt()
+    }
+}
+
+impl Median<f64> for Pareto {
+    /// Returns the median of the Pareto distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// x_m*2^(1/α)
+    /// ```
+    ///
+    /// where `x_m` is the scale and `α` is the shape
+    fn median(&self) -> f64 {
+        self.scale * (2.0_f64.powf(1.0 / self.shape))
```

Comment:
I think you'd be fine with just `2f64` here

Reply:
Sure, but I kind of prefer to be extra-explicit about this being a float literal. It's also consistent then, e.g. when you delete `_f64`, you don't need to add `.0`.

Anyway it's a subjective code style issue, like trailing commas in lists. If you insist on using `2f64`, I'll be glad to change this line as well as ones down below (lines 506-515).

Reply:
While I agree with you on the virtues of explicitness, I believe one would be hard pressed to mistake `f64` for anything other than a float. I honestly don't really feel *that* strongly one way or the other but I would like it to be consistent with the rest of the code base which I believe uses `{n}f64`

Reply:
Sure, will do!

Reply:
Done in 25d501245121570efa4e3f01ea9723b3de5bb044